test/checkout.int.t.sol

```diff
@@ -961,3 +961,141 @@ contract Checkout_buyingWithAlternateERC20_floatingPointPriceMultiple is
     assertEq(_erc20.balanceOf(_market.getNoriFeeWallet()), fee);
   }
 }
+
+contract Checkout_buyingWithCustomFee is Checkout {
+  uint256 ownerPrivateKey = 0xA11CE;
+  address owner = vm.addr(ownerPrivateKey);
+  uint256 customFee = 5;
+  uint256 certificateAmount = 1 ether;
+
+  function setUp() external {
+    _removalIds = _seedRemovals({
+      to: _namedAccounts.supplier,
+      count: 1,
+      list: true
+    });
+    uint256 purchaseAmount = _market.calculateCheckoutTotalWithoutFee(
+      certificateAmount
+    );
+    _market.grantRole({role: _market.MARKET_ADMIN_ROLE(), account: owner});
+    vm.prank(_namedAccounts.admin);
+    _bpNori.deposit(owner, abi.encode(purchaseAmount));
+    vm.prank(owner);
+    _bpNori.approve(address(_market), MAX_INT); // infinite approval for Market to spend owner's tokens
+  }
+
+  function test() external {
+    vm.startPrank(owner);
+    vm.recordLogs();
+    _market.swapWithoutFee(owner, owner, certificateAmount, customFee);
+    vm.stopPrank();
+
+    Vm.Log[] memory entries = vm.getRecordedLogs();
+    bool containsCreateCertificateEventSelector = false;
+    for (uint256 i = 0; i < entries.length; ++i) {
+      if (entries[i].topics[0] == CREATE_CERTIFICATE_EVENT_SELECTOR) {
```

Comment:
this test is problematic because it can pass without ever testing anything if this selector isn't found. The entry index should be deterministic. Can you locate which one it is and use it statically?

Reply:
oh yeah... the test I took this from makes assertions after the loop that the certificate event selector flag was tripped but I must have deleted that. i'll try the static index approach though as it's more clear